test/Attendance.t.sol

```diff
@@ -135,32 +135,95 @@ contract AttendanceTest is Test {
     }
 
     // test non-contract owner cannot create student
-    function testFail_NonOwnerCannotCreateStudent() public pure {
-        assert(false);
-    }
+    function testFail_NonOwnerCannotCreateStudent() public {
+    Student memory student_ = Student({
+        age: 18,
+        attendanceCount: 0,
+        name: student0Name,
+        isRegistered: false,
+        studentAddress: student0Addr
+    });
+
+    // Attempt to create a student using an unauthorized account
+    vm.prank(unauthorizedAcct);
+    attendance.createStudent(student_);
+}
+
 
     // test owner can register a student for a course
-    function test_OwnerCanRegisterStudentForCourse() public pure {
-        assert(true);
-    }
+    function test_OwnerCanRegisterStudentForCourse() public {
+    vm.prank(lecturer);
+
+    // Register a student for CSC201
+    attendance.registerStudentForCourse(CSC201, student0Addr);
+
+    // Verify that the student is registered
+    assertTrue(attendance.isStudentRegisteredForCourse(CSC201, student0Addr));
+}
+
 
     // test a course rep can register a student for a course
-    function test_CourseRepCanRegisterStudentForCourse() public pure {
-        assert(true);
-    }
+    function test_CourseRepCanRegisterStudentForCourse() public {
+    // Authorize a course rep first
+    vm.prank(lecturer);
+    address;
+    courseReps_[0] = rep0;
+    attendance.authorizeCourseRep(CSC201, courseReps_, true);
+
+    // Now, the course rep can register a student
+    vm.prank(rep0);
+    attendance.registerStudentForCourse(CSC201, student0Addr);
+
+    // Verify that the student is registered
+    assertTrue(attendance.isStudentRegisteredForCourse(CSC201, student0Addr));
+}
+
 
     // test a student can de-register themselves from a course
-    function test_StudentCanDeRegisterThemself() public pure {
-        assert(true);
-    }
+    function test_StudentCanDeRegisterThemself() public {
+    vm.prank(lecturer);
+    attendance.registerStudentForCourse(CSC201, student0Addr);
+
+    // Now, the student de-registers themselves
+    vm.prank(student0Addr);
+    attendance.deRegisterFromCourse(CSC201);
+
+    // Verify that the student is no longer registered
+    assertFalse(attendance.isStudentRegisteredForCourse(CSC201, student0Addr));
+}
+
 
     // test authorized reps can increment attendance score
-    function test_CourseRepsCanIncrmentAttendance() public pure {
-        assert(true);
-    }
+    function test_CourseRepsCanIncrmentAttendance() public {
+    // Authorize a course rep first
+    vm.prank(lecturer);
+    address;
```

Comment:
whats the purpose of this line?

Reply:
> whats the purpose of this line?
The line address; within the function test_CourseRepsCanIncrmentAttendance() seems to be mistakenly placed. Didnt notice it imagine, so as it is now, its performing no function. I will refactor the code and push a commit.